### DISCUSSION TITLE
Why is the garbage collection being disabled?

### DISCUSSION THREAD

**Discussion by rudyocalan** [actions/checkout#593] [Q&A]
Hi,

I noticed the checkout action is disabling the garbage collection of the local git repo. Since the .git folders of our local repos on our self-hosted runners are starting to get bigger, I was wondering why it is being disabled and if this is something that should be optional?

https://github.com/actions/checkout/blob/afe4af09a72596f47d806ee5f8b2674ec07fdc73/src/git-source-provider.ts#L92

Thanks for the feedback.

**Comment by t-b** [comment]
Same problem where with self hosted runners.

I guess this is not an issue for the GH runners because they are using short-lived VMs. One workaround I can think of would be to use a post job script [1] and do something like

```
git config gc.auto 1024
git config gc.packLimit 10
git gc --auto
```

in it. But that is also a bit wasteful although this should be fast in the common case that no garbage collection is required.

[1]: https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job#triggering-the-scripts

**Reply by t-b** [reply]
At [1] is an example of a monthly job which runs on all self-hosted runners. The trick is to give a unique label to each runner and then apply a matrix onto it.

[1]: https://github.com/AllenInstitute/MIES/blob/main/.github/workflows/monthly-scheduled.yml
